core/src/main/java/org/infinispan/statetransfer/StateTransferManagerImpl.java

```diff
@@ -286,8 +292,8 @@ public CacheTopology getCacheTopology() {
       }
 
       if (cmdTopologyId < localTopologyId) {
-         ConsistentHash writeCh = cacheTopology.getWriteConsistentHash();
-         Set<Address> newTargets = new HashSet<>(writeCh.locateAllOwners(affectedKeys));
+         Collection<Address> writeTargets = cacheTopology.getWriteTargets(affectedKeys);
```

Comment:
`LCT.getWriteOwners`, never returning `null` would be better. Some people consider `null` an anachronism :)